crates/biome_service/src/file_handlers/json.rs

```diff
@@ -290,13 +290,23 @@ fn lint(params: LintParams) -> LintResults {
 
             let skipped_diagnostics = diagnostic_count - diagnostics.len() as u32;
 
-            let rules = settings.as_rules(params.path.as_path());
-            let rule_filter_list = rules
-                .as_ref()
-                .map(|rules| rules.as_enabled_rules())
-                .unwrap_or_default()
-                .into_iter()
-                .collect::<Vec<_>>();
+            let mut rules = settings.as_rules(params.path.as_path());
+            let rule_filter_list = if let Some(rule) = params.rule {
+                // We execute a single rule because the `--rule` filter is specified.
+                // Set the severity level to its default.
+                if let Some(rules) = rules.as_mut() {
+                    rules.to_mut().set_default_severity(rule.group, rule.name);
+                }
```

Comment:
If we take receive a `&mut` using `as_mut`, why do we need to call `to_mut`?

Reply:
Thank you for the answer. Would you mind going into more details? I am genuinely curious about the technical details 😄

Reply:
In the same function, we compute the final `rules` configuration by taking `overrides` into account. To avoid unnecessary allocation, this computation uses a copy-on-write strategy. Thus, it produces a `Cow<Rules>`.

In the present code we want to override the severity level of the executed rule. We have to mutate `rules`.
We call `to_mut` to turn the Cow into a `Cow::Owned` (if it is not already the case) and get a mutable reference to it.

We could avoid some allocation by also using a copy-on-write strategy. This could require to generate a `set_default_severity` that takes a `Cow<Rules>` instead of `&mut self`.